### PR TITLE
Terraform 0.12+ index can be string or number

### PR DESCRIPTION
**Pull request by sysbot**
In the case of when mixing resources into the statefile in terraform, it's now
possible to have indexes that are not always int. In the example below using the
new 0.12 syntax `foreach` will result in the index being mapped to the
`each.key` name instead of the integer indexes.

```
variable "test" {
  default = {"testa" = "1.1.1.1", "testb" = "2.2.2.2"}
}

resource "aws_route53_record" "this" {
  for_each = var.test
  zone_id = "abc"
  name    = "${each.key}.tv."
  type    = "A"
  ttl     = "300"
  records = [each.value]
}
```

### CODE REVIEW

File: resource.go [hunk 1]
```diff
@@ -125,7 +126,7 @@ func (r Resource) Tags() map[string]string {
 				t[kk] = vv
 			}
 		}
-	case "aws_instance", "linode_instance":
+	case "serverscom_dedicated_server", "aws_instance", "linode_instance":
```
Comment (adammck) [inline resource.go]: This seems to be a separate feature. Could you please split this into a separate PR?

**Comment by adammck** [comment]
Thank you for the patch! Except for the comment above, this looks good to go.

**Comment by sprnza** [comment]
It doesn't work without changes in parser.go. So I get an error like this: `couldn't parse resource keyName: vsphere_virtual_machine.win_slave.win-slave-1.10.178.3.74`

**Comment by atikhono** [comment]
@sprnza Would you please try is my pull request fixed your issue?

@adammck I addressed your comment in my pull request. Could you please review additional changes in resource.go as well? I'm hitting this issue with string index too, would be really happy to have it fixed soon.

**Comment by adammck** [comment]
Merged #139, which includes this. Thanks very much for the original PR. Please re-open this if your issue isn't resolved.
